### DISCUSSION TITLE
Connection string error

### DISCUSSION THREAD

**Discussion by bkristesiashvili** [prisma/prisma#21456] [Q&A] [answered]
`DATABASE_URL="sqlserver://localhost:1433;database=test-web;user=SA;password=abcDEF$;Encrypt=true;TrustServerCertificate=true;"`
 this is my test database connection string, it works when I try to create migrations but when I run app it throws an error
 
 
![image](https://github.com/prisma/prisma-engines/assets/61345934/0317ad55-eb63-45af-9f98-88138cc07035)

as we found out this error caused by the dollar sign symbol into the password.

**Comment by ludralph** [comment, marked as answer]
Hi @bkristesiashvili 👋 

Thank you for raising this question.

Yes, you're correct. The issue you're facing is due to the dollar sign in your password. This is a known issue as documented [here](https://github.com/prisma/prisma/issues/16890).

In your case, the Prisma CLI is able to handle the dollar sign in the password without any issues, but the Prisma Client is not.
To resolve this issue, you should escape the dollar sign in your password using a backslash (\). So your connection string should look like this:
```
DATABASE_URL="sqlserver://localhost:1433;database=test-web;user=SA;password=abcDEF\$;Encrypt=true;TrustServerCertificate=true;"
```
Please note that this solution works with Prisma Client but not with Prisma CLI. So, you might need to adjust the connection string based on whether you're running a Prisma CLI command or a query from Prisma Client.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by bkristesiashvili** [reply]
thank you!

**Comment by levipadre** [comment]
@ludralph Can you help with my connection string too? I can't pull my db.
```
DATABASE_URL="sqlserver://MYCOMPUTERNAME\SQLEXPRESS02;database=Foobar_v10;user=sa;password=}^F*:z934}Qq~:J;trustServerCertificate=true"
```

I assume it's either the server name or the password: `}^F*:z934}Qq~:J`
